configure.ac needs -D_REENTRANT

On Arch Linux, when building from git source, I get plenty of

```
./../apf/apf/cxx_thread_policy.h:31:2: error: #error You need to compile with _REENTRANT defined since this uses threads!                                                 
 #error You need to compile with _REENTRANT defined since this uses threads! 
```

which lets the bulid fail.
`-D_REENTRANT` needs to be added to CPPFLAGS in configure.ac.

--- a/apf/apf/cxx_thread_policy.h
+++ b/apf/apf/cxx_thread_policy.h
@@ -27,10 +27,6 @@
 #ifndef APF_CXX_THREAD_POLICY_H
 #define APF_CXX_THREAD_POLICY_H
 
-#ifndef _REENTRANT
-#error You need to compile with _REENTRANT defined since this uses threads!
-#endif
-
 #ifndef APF_MIMOPROCESSOR_THREAD_POLICY
 #define APF_MIMOPROCESSOR_THREAD_POLICY apf::cxx_thread_policy
 #endif

